pkgs/by-name/up/upscayl/linux.nix

```
@@ -0,0 +1,46 @@
+{
+  appimageTools,
+  makeWrapper,
+
+  pname,
+  version,
+  meta,
+  src,
+}:
+
+let
+  appimageContents = appimageTools.extractType2 {
+    inherit pname version src;
+  };
+in
+appimageTools.wrapType2 {
+  inherit
+    pname
+    version
+    src
+    meta
+    ;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  extraPkgs = pkgs: [
+    pkgs.vulkan-headers
+    pkgs.vulkan-loader
+  ];
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/{applications,pixmaps}
+
+    cp ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
+    cp ${appimageContents}/${pname}.png $out/share/pixmaps/${pname}.png
+
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace-fail 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}'
+
+    wrapProgram $out/bin/${pname} \
```

```suggestion
    cp ${appimageContents}/upscayl.desktop $out/share/applications/upscayl.desktop
    cp ${appimageContents}/upscayl.png $out/share/pixmaps/upscayl.png

    substituteInPlace $out/share/applications/upscayl.desktop \
      --replace-fail 'Exec=AppRun --no-sandbox %U' 'Exec=upscayl'

    wrapProgram $out/bin/upscayl \
```
there is a tracking issue to stop overusing pname

---

@SuperSandro2000 fixed, thanks!